### PR TITLE
v3(enhancement): remove `utils.Trim*` because stdlib has same performance in go1.19

### PR DESCRIPTION
**Pull request by Trim21**
This pr rewrite `Trim`, `TrimRight` and `TrimLeft` to use generic `byteSeq interface{ ~string| ~[]byte }`

### CODE REVIEW

**Comment by ReneWerner87** [comment]
can you check my last comments from the old one ? 
https://github.com/gofiber/fiber/blob/v3-beta/utils/bytes.go#L59

**Review by efectn — COMMENTED** [review]
can you update benchmarks on [README.md](https://github.com/Trim21/fiber/blob/v3-util-genetic-byteSeq/utils/README.md)

**Comment by Trim21** [comment]
do you considering using https://github.com/stretchr/testify instead of current `AssertEqual` 🤔

**Comment by Trim21** [comment]
> can you update benchmarks on [README.md](https://github.com/Trim21/fiber/blob/v3-util-genetic-byteSeq/utils/README.md)

done

**Comment by Trim21** [comment]
I just keep the old tests and benchmark for bytes input

File: ctx.go [hunk 1]
```diff
@@ -1245,7 +1245,7 @@ func (c *DefaultCtx) WriteString(s string) (int, error) {
 // XHR returns a Boolean property, that is true, if the request's X-Requested-With header field is XMLHttpRequest,
 // indicating that the request was issued by a client library (such as jQuery).
 func (c *DefaultCtx) XHR() bool {
-	return utils.EqualFoldBytes(utils.UnsafeBytes(c.Get(HeaderXRequestedWith)), []byte("xmlhttprequest"))
+	return utils.EqualFold(c.Get(HeaderXRequestedWith), "xmlhttprequest")
```
Comment (Trim21) [inline ctx.go:1248]: why do we use `EqualFoldBytes` instead of `EqualFold` in the first place?

**Comment by Trim21** [comment]
I'm running these benchmark in go 1.19. I think we can remove these function now 🤔

```text
Benchmark_TrimRightBytes/fiber-16               100000000               10.66 ns/op            8 B/op          1 allocs/op
Benchmark_TrimRightBytes/default-16             100000000               10.84 ns/op            8 B/op          1 allocs/op
Benchmark_TrimLeftBytes/fiber-16                100000000               10.72 ns/op            8 B/op          1 allocs/op
Benchmark_TrimLeftBytes/default-16              100000000               11.29 ns/op            8 B/op          1 allocs/op
Benchmark_TrimBytes/fiber-16                    74933964                15.43 ns/op           16 B/op          1 allocs/op
Benchmark_TrimBytes/default-16                  74931624                15.86 ns/op           16 B/op          1 allocs/op
Benchmark_TrimRight/fiber-16                    484963700                2.471 ns/op           0 B/op          0 allocs/op
Benchmark_TrimRight/default-16                  454740096                2.614 ns/op           0 B/op          0 allocs/op
Benchmark_TrimLeft/fiber-16                     488699834                2.451 ns/op           0 B/op          0 allocs/op
Benchmark_TrimLeft/default-16                   423590457                2.835 ns/op           0 B/op          0 allocs/op
Benchmark_Trim/fiber-16                         296457481                4.023 ns/op           0 B/op          0 allocs/op
Benchmark_Trim/default-16                       322653328                3.804 ns/op           0 B/op          0 allocs/op
Benchmark_Trim/default.trimspace-16             248519290                4.811 ns/op           0 B/op          0 allocs/op
```

**Comment by efectn** [comment]
> I saw the updated benchmark result, I think we can remove these `Trim*` functions now 🤔
> 
> go 1.19
> 
> ```
> Benchmark_TrimRightBytes/fiber-16               100000000               10.66 ns/op            8 B/op          1 allocs/op
> Benchmark_TrimRightBytes/default-16             100000000               10.84 ns/op            8 B/op          1 allocs/op
> 
> Benchmark_TrimLeftBytes/fiber-16                100000000               10.72 ns/op            8 B/op          1 allocs/op
> Benchmark_TrimLeftBytes/default-16              100000000               11.29 ns/op            8 B/op          1 allocs/op
> 
> Benchmark_TrimBytes/fiber-16                    74933964                15.43 ns/op           16 B/op          1 allocs/op
> Benchmark_TrimBytes/default-16                  74931624                15.86 ns/op           16 B/op          1 allocs/op
> 
> Benchmark_TrimRight/fiber-16                    484963700                2.471 ns/op           0 B/op          0 allocs/op
> Benchmark_TrimRight/default-16                  454740096                2.614 ns/op           0 B/op          0 allocs/op
> 
> Benchmark_TrimLeft/fiber-16                     488699834                2.451 ns/op           0 B/op          0 allocs/op
> Benchmark_TrimLeft/default-16                   423590457                2.835 ns/op           0 B/op          0 allocs/op
> 
> Benchmark_Trim/fiber-16                         296457481                4.023 ns/op           0 B/op          0 allocs/op
> Benchmark_Trim/default-16                       322653328                3.804 ns/op           0 B/op          0 allocs/op
> Benchmark_Trim/default.trimspace-16             248519290                4.811 ns/op           0 B/op          0 allocs/op
> ```
> 
> **1.18** , our version is a little bit faster.
> 
> ```
> goos: windows
> goarch: amd64
> pkg: github.com/gofiber/fiber/v3/utils
> cpu: AMD Ryzen 7 5800X 8-Core Processor
> Benchmark_TrimRightBytes/fiber-16               100000000               10.49 ns/op            8 B/op          1 allocs/op
> Benchmark_TrimRightBytes/default-16             99912576                11.57 ns/op            8 B/op          1 allocs/op
> 
> Benchmark_TrimLeftBytes/fiber-16                100000000               10.42 ns/op            8 B/op          1 allocs/op
> Benchmark_TrimLeftBytes/default-16              99905920                11.55 ns/op            8 B/op          1 allocs/op
> 
> Benchmark_TrimBytes/fiber-16                    70524762                16.44 ns/op           16 B/op          1 allocs/op
> Benchmark_TrimBytes/default-16                  70530980                16.72 ns/op           16 B/op          1 allocs/op
> 
> Benchmark_TrimRight/fiber-16                    865098051                1.398 ns/op           0 B/op          0 allocs/op
> Benchmark_TrimRight/default-16                  487361299                2.470 ns/op           0 B/op          0 allocs/op
> 
> Benchmark_TrimLeft/fiber-16                     760135455                1.569 ns/op           0 B/op          0 allocs/op
> Benchmark_TrimLeft/default-16                   419346405                2.895 ns/op           0 B/op          0 allocs/op
> 
> Benchmark_Trim/fiber-16                         309708110                3.745 ns/op           0 B/op          0 allocs/op
> Benchmark_Trim/default-16                       302284438                4.009 ns/op           0 B/op          0 allocs/op
> Benchmark_Trim/default.trimspace-16             250067830                4.784 ns/op           0 B/op          0 allocs/op
> PASS
> ok      github.com/gofiber/fiber/v3/utils       17.530s
> ```

I agree

**Pull request merged**